packages/cloud-cognitive/src/components/ButtonSetWithOverflow/ButtonSetWithOverflow.js

```diff
@@ -0,0 +1,216 @@
+  };
+
+  useEffect(() => {
+    checkFullyVisibleItems();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [itemArray]);
+
+  const handleResize = () => {
+    // width is the space available for all action bar items horizontally
+    // the action bar items are squares so the height should be one item wide
+    /* istanbul ignore next */ // not sure how to test resize
+    checkFullyVisibleItems();
+  };
+
+  const handleButtonResize = () => {
+    /* istanbul ignore next */ // not sure how to test resize
+    checkFullyVisibleItems();
+  };
+
+  useEffect(() => {
+    if (buttons) {
+      setItemArray(buttons);
+    } else {
+      const unwrapped = unwrapIfFragment(children);
+      setItemArray(
+        unwrapped.map((item) => {
+          const { children: label, ...rest } = item.props;
+          return { label, ...rest };
+        })
+      );
+    }
+  }, [buttons, children]);
+
+  const AButtonSet = React.forwardRef(({ buttons, size, ...rest }, ref) => {
+    return (
+      <ButtonSet {...rest} ref={ref}>
+        {buttons.map(({ label, ...other }, index) => (
+          <Button key={index} {...other} size={size} type="button">
+            {label}
+          </Button>
+        ))}
+      </ButtonSet>
+    );
+  });
+  const ATempComboButton = React.forwardRef(
+    ({ buttons, size, ...rest }, ref) => {
+      return (
+        <TempComboButton
+          {...rest}
+          buttons={buttons}
+          size={size}
+          ref={ref}
+          label={pageActionsLabel}
+        />
+      );
+    }
+  );
+
+  return (
+    <ReactResizeDetector handleWidth={true} onResize={handleResize}>
+      <div className={cx([blockClass, className])} ref={spaceAvailableRef}>
+        <ReactResizeDetector onResize={handleButtonResize}>
+          <div
+            className={`${blockClass}__button-container ${blockClass}__button-container--hidden`}>
+            <AButtonSet
+              aria-hidden={true}
+              ref={sizingContainerRefSet}
+              size={size}
+              buttons={itemArray}
+            />
+          </div>
+        </ReactResizeDetector>
+        <ReactResizeDetector onResize={handleButtonResize}>
+          <div
+            ref={sizingContainerRefCombo}
+            className={`${blockClass}__button-container ${blockClass}__button-container--hidden`}
+            aria-hidden={true}>
+            <ATempComboButton buttons={itemArray} size={size} />
+          </div>
+        </ReactResizeDetector>
+
+        {showAsOverflow ? (
+          <ATempComboButton buttons={itemArray} size={size} />
+        ) : (
+          <AButtonSet
+            className={`${blockClass}__button-container`}
+            size={size}
+            buttons={itemArray}
+          />
+        )}
+      </div>
+    </ReactResizeDetector>
+  );
+};
+
+ButtonSetWithOverflow.propTypes = {
+  /**
+   * Button shape things for us to render - see Carbon button kind, label, onClick.
```

Comment:
```suggestion
   * Button shape things for us to render. Each button is specified as an
   * object with an optional field 'label' to supply the button label.
   * Additional fields in the object will be passed to the Button component,
   * and these can include 'kind', 'onClick', 'className', and any other Button
   * props (but NB not a ref). Any other fields in the object will be passed
   * through to the button element as HTML attributes.
```

NB you could, if you wanted, support passing through a 'ref' in the objects, by using `forwardRef` as you map the objects to Buttons. I did that in ActionSet, in case it was ever handy. But it may not be needed.